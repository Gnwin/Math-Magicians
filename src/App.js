import React from 'react';
import './App.css';
import Calculator from './components/Calculator';

class App extends React.Component() {
  state = {}

  render() {
    return (
      <div className="calculator">
        <Calculator />
      </div>
    );
  }
}

export default App;
