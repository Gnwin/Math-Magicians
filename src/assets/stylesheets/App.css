.nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2rem 0;
  margin-bottom: 3rem;
}

.nav-title {
  font-size: 3rem;
}

.nav-link {
  font-size: 1.5rem;
  padding: 0 0.8rem;
  border-left: 2px solid black;
}

.nav-link:first-child {
  padding-left: 0;
  border-left: none;
}

.nav-link:last-child {
  padding-right: 0;
}

.home-title {
  font-size: 1.7rem;
}

.calcpage {
  display: flex;
  justify-content: space-between;
}

.calctitle {
  font-size: 1.7rem;
}

.calculator {
  display: grid;
  grid-template-columns: repeat(4, auto);
  width: 55%;
}

.answer {
  width: 100%;
  grid-column-start: 1;
  grid-column-end: 5;
  text-align: right;
  padding: 0.7rem 0.6rem 0.7rem 0;
  background-color: rgb(143, 141, 141);
  color: white;
  border: 1px solid white;
}

.zero {
  grid-column-start: 1;
  grid-column-end: 3;
}

.center {
  text-align: center;
  padding: 0.7rem 2.2rem;
  background-color: rgb(227, 227, 227);
  border: 1px solid white;
  cursor: pointer;
}

.orange {
  background-color: rgb(248, 182, 59);
}

.quote {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 55vh;
  padding: 2rem;
}

.quote-msg {
  font-size: 1.7rem;
}
