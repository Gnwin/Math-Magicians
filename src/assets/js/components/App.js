import React from 'react';
import '../../stylesheets/App.css';
import Calculator from './Calculator';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="app">
        <Calculator />
      </div>
    );
  }
}

export default App;
