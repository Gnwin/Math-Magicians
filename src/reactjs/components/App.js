import React, { useState, useEffect } from 'react';
import '../../assets/stylesheets/App.css';
import Calculator from './Calculator';
import numbers from './numbers';
import operands from './operands';

const App = () => {
  const [calcNumbers, setNumbers] = useState([]);
  const [calcOperands, setOperands] = useState([]);

  useEffect(() => {
    setNumbers(numbers);
    setOperands(operands);
  }, []);

  return (
    <div className="app">
      <Calculator operands={calcOperands} numbers={calcNumbers} />
    </div>
  );
};

export default App;
