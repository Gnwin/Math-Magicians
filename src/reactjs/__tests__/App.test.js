import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

import App from '../components/App';

describe('it tests the App Component', () => {
  it('renders the App Component', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    const appComponent = screen.getByTestId('app-1');
    expect(appComponent).toBeInTheDocument();
  });

  it('renders quote component correctly', () => {
    const tree = renderer
      .create(<BrowserRouter><App /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
