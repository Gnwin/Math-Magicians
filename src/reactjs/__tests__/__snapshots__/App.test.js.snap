// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the App Component renders A component correctly 1`] = `
<div
  className="appwidth"
  data-testid="app-1"
>
  <div
    className="header"
  >
    <div
      className="nav"
      data-testid="nav-a"
    >
      <div
        className="nav-title"
      >
        Math magicians
      </div>
      <div
        className="nav-links"
        data-testid="nav-b"
      >
        <a
          className="nav-link"
          data-testid="nav-1"
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-testid="nav-2"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <a
          className="nav-link"
          data-testid="nav-3"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
  </div>
  <div
    className="main"
  >
    <div
      data-testid="home-1"
    >
      <h1
        className="home-title"
        data-testid="home-2"
      >
        Welcome to our page!
      </h1>
      <br />
      <p
        data-testid="home-3"
      >
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores.
      </p>
      <br />
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores.
      </p>
      <br />
    </div>
  </div>
</div>
`;

exports[`it tests the App Component renders App component correctly 1`] = `
<div
  className="appwidth"
  data-testid="app-1"
>
  <div
    className="header"
  >
    <div
      className="nav"
      data-testid="nav-a"
    >
      <div
        className="nav-title"
      >
        Math magicians
      </div>
      <div
        className="nav-links"
        data-testid="nav-b"
      >
        <a
          className="nav-link"
          data-testid="nav-1"
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-testid="nav-2"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <a
          className="nav-link"
          data-testid="nav-3"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
  </div>
  <div
    className="main"
  >
    <div
      data-testid="home-1"
    >
      <h1
        className="home-title"
        data-testid="home-2"
      >
        Welcome to our page!
      </h1>
      <br />
      <p
        data-testid="home-3"
      >
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores.
      </p>
      <br />
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores.
      </p>
      <br />
    </div>
  </div>
</div>
`;

exports[`it tests the App Component renders quote component correctly 1`] = `
<div
  className="appwidth"
  data-testid="app-1"
>
  <div
    className="header"
  >
    <div
      className="nav"
      data-testid="nav-a"
    >
      <div
        className="nav-title"
      >
        Math magicians
      </div>
      <div
        className="nav-links"
        data-testid="nav-b"
      >
        <a
          className="nav-link"
          data-testid="nav-1"
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-testid="nav-2"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <a
          className="nav-link"
          data-testid="nav-3"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
  </div>
  <div
    className="main"
  >
    <div
      data-testid="home-1"
    >
      <h1
        className="home-title"
        data-testid="home-2"
      >
        Welcome to our page!
      </h1>
      <br />
      <p
        data-testid="home-3"
      >
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores.
      </p>
      <br />
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officiis cumque explicabo iusto magnam harum aut. Eum porro neque, nostrum cupiditate delectus minus cumque ea laboriosam tempora libero laborum quos dolorem! Odio consectetur ex eum. Dignissimos, laboriosam? Nobis incidunt animi ullam sed modi dolor error rerum aperiam facilis. Cumque eaque maiores corporis odio, illo quos? Saepe laboriosam quis dolores ratione asperiores.
      </p>
      <br />
    </div>
  </div>
</div>
`;
